test/domain/action/action.test.ts

```diff
@@ -103,6 +103,101 @@ describe('Action', () => {
         })
       })
     })
+
+    describe('updateAction', () => {
+      describe('quand le statut est fourni', () => {
+        it("renvoie l'action avec le statut et la date à jour", async () => {
+          // Given
+          const action = uneAction({
+            statut: Action.Statut.PAS_COMMENCEE
+          })
+
+          const enCours = Action.Statut.EN_COURS
+
+          const infosActionAMettreAJour = {
+            idAction: 'id-action',
+            statut: enCours,
+            description: 'Une superbe description'
+          }
+
+          // When
+          const resultAction = actionFactory.updateAction(
+            action,
+            infosActionAMettreAJour
+          )
+
+          // Then
```

Comment:
Attention ici tu testes plusieurs comportements qui pourraient se retrouver dans des 'it' différents. Le côté actualisation de la date est une règle métier différente de juste dire l'attribut demandé est bien modifié (d'ailleurs la date d'actualisation est pas demandée fonctionnellement c'est nous pour le support pour l'instant ça me choquerait même pas qu'elle soit pas testée unitairement)

Reply:
et du coup un expect( isSuccess(result) && result.data.statut ).to.equal( enCours ) ça marche plutôt qu'avoir un expect dans un if

Reply:
Un exmeple d'arborescence de describe comme je décrivais plus haut :
1 - On réussit à modifier
- 1a - maj l'attribut en question
- 1b - maj la dateFinRelle si le statut est à Terminée
- 1c - maj la date actualisation
2 - On réussit pas à modifier